front/src/applications/operationalStudies/components/ManageTrainSchedule/SpeedLimitByTagSelector.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { ComponentType, useState, useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useTranslation } from 'react-i18next';
+import { updateSpeedLimitByTag } from 'reducers/osrdconf';
+import { setFailure } from 'reducers/main';
+import { get } from 'common/requests';
+import icon from 'assets/pictures/components/speedometer.svg';
+import SelectImprovedSNCF from 'common/BootstrapSNCF/SelectImprovedSNCF';
+import { getInfraID, getSpeedLimitByTag } from 'reducers/osrdconf/selectors';
+import { Dispatch } from 'redux';
+
+interface getListByTagFn {
+  (): Promise<string[]>;
+}
+
+type SpeedLimitByTagSelectorProps = {
+  condensed?: boolean;
+  infraID?: number;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  speedLimitByTag?: any;
+  dispatch?: Dispatch;
+  getTagsList?: getListByTagFn;
+};
+
+function withOSRDData<T>(Component: ComponentType<T>) {
+  return (hocProps: T) => {
+    const { t } = useTranslation(['operationalStudies/manageTrainSchedule']);
+    const dispatch = useDispatch();
+    const infraID = useSelector(getInfraID);
+    const speedLimitByTag = useSelector(getSpeedLimitByTag);
+    const getTagsList = async (): Promise<T> => {
+      let tagList = [];
+      try {
+        tagList = await get(`/editoast/infra/${infraID}/speed_limit_tags/`);
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
+      } catch (e: any) {
+        dispatch(
+          setFailure({
+            name: t('errorMessages.unableToRetrieveTags'),
+            message: `${e.message} : ${e.response && e.response.data.detail}`,
+          })
+        );
+      }
+      return tagList;
+    };
+    return (
+      <Component
+        {...(hocProps as T)}
+        dispatch={dispatch}
+        infraID={infraID}
+        speedLimitByTag={speedLimitByTag}
+        getTagsList={getTagsList}
+      />
+    );
+  };
+}
+
+export function SpeedLimitByTagSelector({
+  infraID,
+  speedLimitByTag,
+  dispatch = () => {},
+  getTagsList = async (): Promise<Array<string>> => [],
+  condensed = false,
+}: SpeedLimitByTagSelectorProps) {
+  const [speedLimitsTags, setSpeedLimitsTags] = useState<any[] | undefined>(undefined);
+  const [oldInfraID, setOldInfraID] = useState(infraID);
```

Comment:
```suggestion
  const oldInfraID = useRef(infraID);
```
You can use a ref here rather than a state value :) This way, when the value change, it will not trigger a new re-render
https://react.dev/reference/react/useRef

Reply:
done

Reply:
And finally removed in flavor of RTK logic